tnb/banks.py

```diff
@@ -185,12 +190,12 @@ def connection_requests(self, node_id, signature):
 
     def post_invalid_block(
```

Comment:
Shouldn't we add some type like `Signature`, `Signing_key`, `Identifier` ?
Just to make sure that we will not pass Signink_key where a Signature must be sent for example.
For now, everything is `str` but it could be dangerous if someone put the wrong argument.

Reply:
I agree that there should be some validation for the different keys, but this will need a more elaborate discussion. This change will probably introduce some breaking changes, so I would suggest opening a new issue with a proper proposal for this.

Reply:
I'm not talking about creating class for signing keys/node identifier, but simply use an important feature of `typing` which is to create type aliases. No change in the code would be needed but simply replace 
`post_invalid_block(signing_key: str, signature: str, nid:str)` by `post_invalid_block(signing_key: SigningKey,  signature: Signature, nid:NodeIdentifier)`

Reply:
I understand exactly what you meant. The value of `typings` lies in that there is some sort of validation for the parameters. 
Currently the parameters you named all will have a value of the `same` type, so there is no way to differentiate between the values at the moment. 

This means that even if you somehow are able to give them these typing names, you can still be able to switch up the values by mistake without the system warning you. 

Creating classes would be a solution to this problem, but like I said before this would need to be discussed with the team as a whole. Hence I suggested for a new issue.